expression/scalar_function.go

```diff
@@ -191,6 +211,17 @@ func (sf *ScalarFunction) Eval(row []types.Datum) (d types.Datum, err error) {
 			res, isNull, err = sf.EvalString(row, sc)
 		}
 	}
+
+	if err != nil && terror.ErrorEqual(err, types.ErrOverflow) &&
+		sf.FuncName.L == ast.UnaryMinus && !sc.InUpdateOrDeleteStmt {
+		// TODO: fix #3762, overflow convert it to decimal
+		err = sf.convertArgsToDecimal(sc)
```

Comment:
no, mysql did not have any document refer to it, but there is some relevent doc: https://dev.mysql.com/doc/refman/5.7/en/out-of-range-and-overflow.html,